charms/worker/k8s/src/token_distributor.py

```diff
@@ -246,14 +257,17 @@ def recover_token(self, relation: ops.Relation) -> Generator[str, None, None]:
             if (secret_id := relation.data[unit].get(secret_key))
         }
 
-        assert len(secret_ids) == 1, f"Failed to find 1 {relation.name}:{secret_key}"  # nosec
+        if len(secret_ids) != 1:
+            raise ReconcilerError(f"Failed to find 1 {relation.name}:{secret_key}")
         (secret_id,) = secret_ids
-        assert secret_id, f"{relation.name}:{secret_key} is not valid"  # nosec
+        if not secret_id:
+            raise ReconcilerError(f"{relation.name}:{secret_key} is not valid")
         secret = self.charm.model.get_secret(id=secret_id)
 
         # Get the content from the secret
         content = secret.get_content(refresh=True)
-        assert content["token"], f"{relation.name}:token not valid"  # nosec
+        if content.get("token") is None:
+            raise ReconcilerError(f"{relation.name}:token not valid")
```

Comment:
i think this is more accurate

```suggestion
        if not content.get("token"):
            raise ReconcilerError(f"{relation.name}:token not valid")
```

Reply:
Thanks! Fixed this. Just as a question: Are we changing this to also include content.get("token") == ""?